src/doc/en/developer/coding_basics.rst

```diff
@@ -153,17 +153,57 @@ Adding new top-level packages below :mod:`sage` should be done
 sparingly.  It is often better to create subpackages of existing
 packages.
 
-Non-Python Sage source code and supporting files can be included in one
-of the following places:
+Non-Python Sage source code and small supporting files can be
+included in one of the following places:
 
 - In the directory of the Python code that uses that file.  When the
   Sage library is installed, the file will be installed in the same
-  location as the Python code. For example,
-  ``SAGE_ROOT/src/sage/interfaces/maxima.py`` needs to use the file
-  ``SAGE_ROOT/src/sage/interfaces/maxima.lisp`` at runtime, so it refers
-  to it as ::
+  location as the Python code. This is referred to as "package data".
+
+  The preferred way to access the data from Python is using the
+  `importlib.resources API <https://importlib-resources.readthedocs.io/en/latest/using.html>`,
+  in particular the function :func:`importlib.resources.files`.
+  It should be imported as follows
+  (see :ref:`section-python-language-standard`)::
+
+    try:
+        # Use backport package providing Python 3.11 features
+        from importlib_resources import files
+    except ImportError:
+        from importlib.resources import files
```

Comment:
```suggestion
    from importlib.resources import files
```

maybe with a note about not using resources in namespace packages until 3.9 is removed support, i.e. apr 2024?. Really not worth doing this for 2 months and using the version of importlib.resources from python is better to avoid bleeding edge and stick to long term stable API.

Your code allows one, in principle, to use features from new `importlib_resources` which are not available in the version of python used. But if those features are indeed used, this results in an incompatibility with a system where python is new enough to have `importlib.resources` but where `importlib_resources` is not installed (or not up to date). From that point of view is much better to stick to the stable API, meaning features that are common available in all the supported versions of python.

The only newer nicety I see (available on 3.12) is that one can use `files()` to stand for `files(__package__)`

Reply:
The case of "`importlib_resources` not up to date" is a real point. I'll make a change.

I don't agree with the rest what you're saying.

Reply:
In abf3ae6, I've changed it to using an explicit version check (for 3.11!) instead of `try... except`.

I'm using 3.11, not 3.9. 
Note the section on "Language Standard" in the Developer's Guide: The idea is that we develop the Sage library to one language standard, and that's currently 3.9 + the backports for 3.11.

Reply:
> until 3.9 is removed support, i.e. apr 2024?

We don't have a set schedule for this yet.

Reply:
> The only newer nicety I see (available on 3.12) is that one can use files() to stand for files(__package__)

That's quite nice, so let's please merge
- #37402

Reply:
> > until 3.9 is removed support, i.e. apr 2024?
> 
> We don't have a set schedule for this yet.

42 months as per NEP 29 is apr 2024. They call this best practices.

Reply:
Please @tornaria. Our project has not adopted NEP 29. If you want to know about that, read
 https://github.com/sagemath/sage/wiki/NEP-29:-Python-version-strategy